### DISCUSSION TITLE
[Linux]Perfomance hit since march releases

### DISCUSSION THREAD

**Discussion by nromn** [ppy/osu#23802] [Q&A] [answered]
### Type

Performance

### Bug description

Since march releases I had a significant FPS drop on my machine (from 700~FPS average to bottom spikes down to 70, with consistent stutter). My laptop has a Intel i7-8750H and an NVIDIA GeForce 1050 mobile (I'm using the proprietary drivers from NVIDIA). Tested this with two distros, OpenSUSE Tumbleweed and Fedora 38, both using Gnome 44 (also tried switching between x11 and Wayland and the Unofficial Flatpak and AppImage packages). I Don't have any idea on what could be causing it, outside of the renderer (tried OpenGL legacy and the new one too), since the vulkan one isn't available anymore. If any more information I could provide would help to diagnose/solve the problem, just ask for it and I'll try my best to help.

### Version

2023.326.0 - 2023.605.0

### Logs

[database.log](https://github.com/ppy/osu/files/11677523/database.log)
[input.log](https://github.com/ppy/osu/files/11677524/input.log)
[network.log](https://github.com/ppy/osu/files/11677525/network.log)
[performance.log](https://github.com/ppy/osu/files/11677526/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11677527/runtime.log)

**Comment by fyyrraaa** [comment]
Not a maintainer, but could it be that osu! chose the Intel integrated graphics instead of the Nvidia one? That might be the reason why it ran poorly.

**Reply by nromn** [reply]
Well, after looking a bunch into it, I believe the issue really lies into this. I believe this is not the place for getting support, so I'll try some linux forums out there (for those with a similar problem, apparently hybrid graphics cards are not fun to make work with linux). And yes, my whole system isn't using my graphics dedicated GPU anymore. That will be a bless of a week. Thank you guys for the support, and keep up with the amazing job in developing osu!.

**Reply by nromn** [reply, marked as answer]
Well, tried the obvious. Just uninstalled the NVIDIA drivers and reinstalled them (made sure this time secure boot was disabled) via GNOME software, nonetheless. Now my problem is how to make the game run using it, cause everything else seems to be running it ok, even the "about" section recognizes my two cards. I don't have a clue about how to make this either on flatpak or AppImage. My FPS is just marginally better in menus, but in game it runs badly still.

**Reply by fyyrraaa** [reply]
I found this, since you said that you ran openSUSE: https://en.opensuse.org/SDB:NVIDIA_SUSE_Prime

Alternatively, you might wanna have a look at the ArchWiki for some pointers (https://wiki.archlinux.org/title/PRIME#PRIME_render_offload and https://wiki.archlinux.org/title/NVIDIA_Optimus). I myself don't have a hybrid graphics setup but hopefully it'll point you somewhere.
